### PR TITLE
we create a protocol and a struct wich it bring us data

### PR DESCRIPTION
**Pull request by JonathanAmador12**
Create networking layer.
All logic related to http request is there.

### CODE REVIEW

File: PokedexV2/Networking/APIError.swift [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+//
+//  Created by JonathanA on 09/10/22.
+//
+
+import Foundation
+
+enum ApiError: Error {
+    case badDecoding
+    case badUrl
+    case noAutorize
```
Comment (AmadorHeE) [inline PokedexV2/Networking/APIError.swift]: It should be 'unauthorized'

File: PokedexV2/Networking/HTTPRequestData.swift [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    }
+    var queryParams: [String:String]{
+        [:]
+    }
+    var headers: [String:String]{
+        [:]
+    }
+    
+}
+// we need to create athother extension for we can implement a fucion like a astract funcion
```
Comment (AmadorHeE) [inline PokedexV2/Networking/HTTPRequestData.swift]: Change this.

**Review by AmadorHeE — CHANGES_REQUESTED** [review]

File: PokedexV2/Networking/APIError.swift (hunk 1 above)
Comment (JonathanAmador12) [reply PokedexV2/Networking/APIError.swift]: fixed

File: PokedexV2/Networking/HTTPRequestData.swift (hunk 2 above)
Comment (JonathanAmador12) [reply PokedexV2/Networking/HTTPRequestData.swift]: fixed

**Review by AmadorHeE — APPROVED** [review]

**Pull request merged**